aztec/src/main/kotlin/org/wordpress/aztec/AztecText.kt

```diff
@@ -446,6 +443,207 @@ class AztecText : EditText, TextWatcher {
         }
 
     }
+
+    // HeaderSpan ==================================================================================
+
+    fun header(format: Boolean, textFormat: TextFormat) {
+        headerClear()
+
+        if (format) {
+            headerFormat(textFormat)
+        }
+    }
+
+    private fun headerClear() {
+        val lines = TextUtils.split(editableText.toString(), "\n")
+
+        for (i in lines.indices) {
+            if (!containHeader(i)) {
+                continue
+            }
+
+            var lineStart = 0
+
+            for (j in 0..i - 1) {
+                lineStart += lines[j].length + 1
+            }
+
+            val lineEnd = lineStart + lines[i].length
+
+            if (lineStart >= lineEnd) {
+                continue
+            }
+
+            var headerStart = 0
+            var headerEnd = 0
+
+            if (lineStart <= selectionStart && selectionEnd <= lineEnd) {
+                headerStart = lineStart
+                headerEnd = lineEnd
+            } else if (selectionStart <= lineStart && lineEnd <= selectionEnd) {
+                headerStart = lineStart
+                headerEnd = lineEnd
+            }
+
+            if (headerStart < headerEnd) {
+                val spans = editableText.getSpans(headerStart, headerEnd, AztecHeaderSpan::class.java)
+
+                for (span in spans) {
+                    editableText.removeSpan(span)
+                }
+
+                refreshText()
```

Comment:
Do you think we can put this method outside of the loop? Maybe we can call it once after spans are removed from each line.